instrumentation/opentelemetry-api-beta/src/test/groovy/ContextBridgeTest.groovy

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import application.io.opentelemetry.OpenTelemetry
+import application.io.opentelemetry.context.Context
+import application.io.opentelemetry.context.ContextKey
+import application.io.opentelemetry.trace.Span
+import io.opentelemetry.instrumentation.test.AgentTestRunner
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import java.util.concurrent.atomic.AtomicReference
+import spock.lang.Shared
+
+class ContextBridgeTest extends AgentTestRunner {
+
+  private static final ContextKey<String> ANIMAL = ContextKey.named("animal")
+
+  private static final io.opentelemetry.context.ContextKey<String> FOOD =
+    io.opentelemetry.context.ContextKey.named("food")
+  private static final io.opentelemetry.context.ContextKey<String> COUNTRY =
+    io.opentelemetry.context.ContextKey.named("country")
+
+  @Shared
+  ExecutorService otherThread
+
+  def setupSpec() {
+    otherThread = Executors.newSingleThreadExecutor()
+  }
+
+  def cleanupSpec() {
+    otherThread.shutdown()
+  }
+
+  def "agent and application mix"() {
```

Comment:
does this test show interop, or just side-by-side, since everything is in the same thread?

would it be better to `Context.root().makeCurrent().withCloseable {}` (for both agent and non-agent) around each existing `makeCurrent().withCloseable {}` (except for the first) below?

Reply:
I don't think this one shows interop indeed, just the context parenting. The below tests have "another thread" for propagation test.

Not sure I understand `would it be better to Context.root().makeCurrent().withCloseable {}` - if we mount root in the middle of the tree it's like resetting context, right?

Reply:
ya, you're right, the test below addresses propagating across threads, so no need for that in this test